### ISSUE TITLE
This is mainly a question rather than an issue

### ISSUE THREAD

**Issue by shadow111** [innat/TensorFlow-2-Practice#1]
After saving the model using
`model.save('net',save_format='tf')`
and reloading it
`re_load_model = tf.keras.models.load_model("net", compile=False)`

How we can resume training?
Thanks in advance

**Comment by innat** [comment]
Same as you train before. 

```
model.save('net',save_format='tf')
re_load_model = tf.keras.models.load_model("net", compile=True)
re_load_model.fit(..)
```

**Comment by shadow111** [comment]
yes I understand that , but in the case of a custom training loop we are not using the fit method  as your notebook here 
https://github.com/innat/TensorFlow-2-Practice/blob/main/Custom_Loop_Training_InceptionNet.ipynb
after we load the model how we can resume Training with custom training loop as used in the notebook please.
Thanks

**Comment by innat** [comment]
After loading the saved weights, just re-run [this](https://nbviewer.jupyter.org/github/innat/TensorFlow-2-Practice/blob/main/Custom_Loop_Training_InceptionNet.ipynb#Custom-Loop-Training-from-Scratch) cell. Your training progress will start where you left. But you probably need some mechanism to track the last learning rate like stuff.

**Comment by shadow111** [comment]
okay do you have any recommendation how to track the last learning rate and why? it may sound a simple question but i need your help please

**Comment by innat** [comment]
In a custom training loop, all you need to implement by yourself, ie. various callback functions that are already exist in `keras`. I would recommend not to use a custom training loop in case you don't have any intention to write various callback functionalities yourself. 

Instead you should use `.fit` and if you want to override the `train_step` function then [customize the `.fit` method.  ](https://keras.io/guides/customizing_what_happens_in_fit/). By that, you are able to use [callback methods](https://keras.io/api/callbacks/) and also modify the `.fit` method as you did in the custom training loop. 

It's matters because when you use callback you probably use various functions like `ReduceLROnPlateau`, `CSVLogger` - by that you can track down the last epoch, lat learning rate, etc. And when you resume the model training, all you need to set the last learning rate and use `model.fit(..., initial_epoch=0)`. Here 

```
initial_epoch - 
An Integer. Epoch at which to start training (useful for 
resuming a previous training run).
```

## Example 

Override the `train_step` method, more details are [here](https://keras.io/guides/customizing_what_happens_in_fit/).

```
class CustomModel(keras.Model):
    def train_step(self, data):
        # Unpack the data. Its structure depends on your model and
        # on what you pass to `fit()`.
        x, y = data
        with tf.GradientTape() as tape:
            y_pred = self(x, training=True)  # Forward pass
            # Compute the loss value
            # (the loss function is configured in `compile()`)
            loss = self.compiled_loss(y, y_pred, regularization_losses=self.losses)
        # Compute gradients
        trainable_vars = self.trainable_variables
        gradients = tape.gradient(loss, trainable_vars)
        # Update weights
        self.optimizer.apply_gradients(zip(gradients, trainable_vars))
        # Update metrics (includes the metric that tracks the loss)
        self.compiled_metrics.update_state(y, y_pred)
        # Return a dict mapping metric names to current value
        return {m.name: m.result() for m in self.metrics}
```

Run the model 

```
# Construct and compile an instance of CustomModel
inputs = keras.Input(shape=(32,))
outputs = keras.layers.Dense(1)(inputs)
model = CustomModel(inputs, outputs)
model.compile(optimizer="adam", loss="mse", metrics=["mae"])

# Just use `fit` as usual
x = np.random.random((1000, 32))
y = np.random.random((1000, 1))
model.fit(x, y, epochs=3, callbacks=[
                          tf.keras.callbacks.ReduceLROnPlateau(monitor="val_loss",patience=2,
                                   mode="auto", cooldown=0, min_lr=0),
                         tf.keras.callbacks.CSVLogger('log.csv')
                  ])

model.save_weights('wg.h5')
```

Next, while resume training again, we can get the last epoch and last `lr` from that `CSV` file. And resume training as follows 

```
opt = keras.optimizers.Adam(learning_rate= LAST_LEARNING_RATE_CSV_FILE)
model = CustomModel(inputs, outputs)
model.laod_weights('wg.h5')
model.compile(optimizer=opt, loss="mse", metrics=["mae"])
model.fit(x, y, epochs=5, initial_epoch = LAST_EPOCH_NUUMBER_CSV_FILE)
```
